File: all_banchmark/03_sum_squares/sum_squares.c
// Sum of Squares Benchmark - C
// Simple arithmetic loop to test basic operations

#include <stdio.h>
#include <stdint.h>

int main() {
    int64_t limit = 100000000;  // 100 million
    int64_t sum = 0;
    
    for (int64_t i = 1; i <= limit; i++) {
        sum += i * i;
    }
    
    printf("Sum of squares 1 to %lld = %lld\n", limit, sum);
    return 0;
}

File: all_banchmark/03_sum_squares/sum_squares.rs
// Sum of Squares Benchmark - Rust
// Simple arithmetic loop to test basic operations

fn main() {
    let limit: i64 = 100_000_000;  // 100 million
    let mut sum: i64 = 0;
    
    let mut i: i64 = 1;
    while i <= limit {
        sum += i * i;
        i += 1;
    }
    
    println!("Sum of squares 1 to {} = {}", limit, sum);
}
